From c710b2bf095e9301066d33a13e8a406f0d134b2f Mon Sep 17 00:00:00 2001
From: Arnaud Lefebvre <a.lefebvre@outlook.fr>
Date: Tue, 30 May 2017 13:45:32 +0200
Subject: [PATCH 1/1] nmtui connect: avoid segfault when iface is not found

https://github.com/NetworkManager/NetworkManager/pull/21
(cherry picked from commit 1fcbb69ae22ed4a6047e36c816f49b2a67a13583)
(cherry picked from commit cc36b9f6eb7870d02e0df7c26a7a980709b620fd)
---
 clients/tui/nmt-connect-connection-list.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/clients/tui/nmt-connect-connection-list.c b/clients/tui/nmt-connect-connection-list.c
index edfaa09..0011fc5 100644
--- a/clients/tui/nmt-connect-connection-list.c
+++ b/clients/tui/nmt-connect-connection-list.c
@@ -621,6 +621,7 @@ nmt_connect_connection_list_get_connection (NmtConnectConnectionList  *list,
 	NmtConnectDevice *nmtdev;
 	NmtConnectConnection *nmtconn = NULL;
 	NMConnection *conn = NULL;
+	const char *iface = NULL;
 
 	g_return_val_if_fail (identifier, FALSE);
 
@@ -643,9 +644,12 @@ nmt_connect_connection_list_get_connection (NmtConnectConnectionList  *list,
 				goto found;
 		}
 
-		if (!conn && nmtdev->device && !strcmp (identifier, nm_device_get_ip_iface (nmtdev->device))) {
-			nmtconn = nmtdev->conns->data;
-			goto found;
+		if (!conn && nmtdev->device) {
+			iface = nm_device_get_ip_iface (nmtdev->device);
+			if (iface && !strcmp (identifier, iface)) {
+				nmtconn = nmtdev->conns->data;
+				goto found;
+			}
 		}
 	}
 
-- 
2.9.4

